### DISCUSSION TITLE
Description large repeated sequences

### DISCUSSION THREAD

**Discussion by jtdendunnen** [HGVSnomenclature/hgvs-nomenclature#81] [HGVS nomenclature proposals]
The current format to describe variants in repeated sequences has the format g.123_152CAG[10] where "123_152" describes the segment of the reference sequence containing the repeated sequence, "CAG" the repeated sequence of one repeat unit, and "[10]" the number of repeated units. When the sequence of a repeat unit becomes larger this format becomes rather cumbersome, especially when, as often the case, the copies of the repeat unit are not perfectly identical. Take e.g. the NEB gene which has an imprefect 8-exon repeat (exon 82-89, 90-97 and 98-105) in the reference sequence. In individuals with nemaline myopathy this can be extended up to 11 units. HGVS nomenclature has currently no format to describe such variants.

Proposal:  the format g.12300_12899rep[3>11] can be used to describe a 200 nucleotide repeat where "12300_12899" describes the location of the repeated sequence, "rep" indicates the presence of a repeated segment, "N[3>11]" describes the variant which has 3 units in the reference sequence and is substituted by 11 units in the variant sequence. An alternative would be to describe the repeated unit as "(N[(200)])", i.e. a unit of 200 not specified nucleotides ("N") but this makes it more complex and not correct since often the length of the repeat unit will differ as well (giving "(N[(198_204)])").

**Comment by ahwagner** [comment]
I like the proposed solution. We come up with a data class in VRS that handles this concept in much the same way. That class also allows for the description of repeat _ranges_, which could be added to this proposed syntax as:

```
g.12300_12899rep[3>11,14]
```
In this example, the `[3>11,14]` segment describes the variant having 3 units in the reference sequence substituted by 11 to 14 units in the variant sequence. In this model, the `3` is analogous to the VRS [RepeatSubunitLength](https://github.com/ga4gh/vrs/blob/0c9498d4eae1e868ac0b49b7e480fd6a3d67468c/schema/vrs-source.yaml#L482-L485) and `11,14` analogous to the [Length](https://github.com/ga4gh/vrs/blob/0c9498d4eae1e868ac0b49b7e480fd6a3d67468c/schema/vrs-source.yaml#L473-L481) field using a [Range](https://github.com/ga4gh/vrs/blob/0c9498d4eae1e868ac0b49b7e480fd6a3d67468c/schema/vrs-source.yaml#L660-L670) object.

The proposed syntax would make for an easy transposition between HGVS and VRS for these types of expressions!

**Reply by ifokkema** [reply]
I agree with Johan that ranges are indicated using underscores. Uncertain ranges are indicated with parentheses as well, i.e., `(11_14)`.

**Comment by jtdendunnen** [comment]
In this example, the [3>11,14] segment describes the variant having 3 units in the reference sequence substituted by 11 to 14 units in the variant sequence.

Note that following HGVS recommendations this should be g.12300_12899rep[3>(11_14)]

**Comment by ifokkema** [comment]
In my opinion, the main problem with this proposal is that it creates an alternative syntax for something that can be described using the current nomenclature. An extension of the repeat, when sequenced, can already be described correctly using the repeat syntax. Also using `dup` or `ins` syntax. Now we would add yet another description for the same variant. Its usefulness would only apply to variants where the exact sequence of the insertion isn't known. But then, we can also already use the `ins` syntax. Having more rules about which description type should take preference over other description types makes the HGVS nomenclature increasingly complex.

> An alternative would be to describe the repeated unit as "(N[(200)])", i.e. a unit of 200 not specified nucleotides ("N") but this makes it more complex and not correct since often the length of the repeat unit will differ as well (giving "(N[(198_204)])").

I personally find `g.(12300_12899)insN[(1600_2200)]` less complex than `g.12300_12899rep[3>(11_14)]`, but that's probably because I know HGVS nomenclature syntax.

The only thing is that the latter suggests the inserts are multiples of 200 bases, whereas the former description could also mean an insertion of 1601 bases. However, since clearly, the inserted sequence hasn't been sequenced, claiming that the insertion is an extension of the repeat is a mere assumption so `rep` would be incorrect.

**Reply by jtdendunnen** [reply]
Please note two things;
- while from g.12300_12899rep[3>(11_14)] I can immediately conclude the variant has 11 to 14 copies of a repeat unit, I can not conclude this from g.(12300_12899)insN[(1600_2200)]. There is e.g. nothing in g.(12300_12899)insN[(1600_2200)] indicating there is a 200 nt repeat unit in this section.
- "claiming that the insertion is an extension of the repeat is a mere assumption so rep would be incorrect". I do not agree. Usually the repeat units are determined from hybridization intensity (arrays), PCR and restriction enzyme digestion and sizing of units/full length fragment size or optical mapping.
